### DISCUSSION TITLE
[4.x] Feature request - Allow duplicate email addresses for users

### DISCUSSION THREAD

**Discussion by anchovy** [craftcms/cms#14577] [Ideas] [answered]
What's the chance of allowing duplicate email addresses for users?

On the last two site's we developed we've had cases where the same email address needed to be used for different users (always with a unique username!).

Some of the reasons behind this are...
1: The user doesn't have an email address at all (crazy but true), in this case we assign a random email address.
2: Some institutional users have the same email address i.e. Husband & Wife, but they need separate user accounts.
3: Some users need to have 2x accounts for different purposes, but they can't belong to 2x User groups for various reasons.

We've got around this by using the User::EVENT_BEFORE_SAVE event and prefixing a unique id to the email i.e. xx_test@test.com and yy_test@test.com and then removing this on the ElementQuery::EVENT_AFTER_POPULATE_ELEMENT, so both users have the 'test@test.com' email address when loaded.

If we had a craft option of allowing duplicate email addresses we wouldn't need all this extra processing.

Any chance of this @brandonkelly ?

**Comment by brandonkelly** [comment, marked as answer]
Zero chance. If two users shared the same email address, they would have the ability to log in with each others’ accounts via the reset-password workflow, by entering the other user’s username.

**Reply by anchovy** [reply]
Resetting a password via entering their username would trigger a rest password against the matching username/user, not their email? We'd just need to disable matching resetting using email address as the key if duplicate emails was enabled?

**Reply by brandonkelly** [reply]
Sorry, there’s just no way this would be considered. Users must have a unique email address.
